### ISSUE TITLE
Failed resolution of firebasemessingservice on Android

### ISSUE THREAD

**Issue by chrisevans9629** [Microsoft/AppCenter-SDK-DotNet#826]
### **Description**
I'm getting this error on startup:
Java.Lang.NoClassDefFoundError: failed resolution of Lcom/google/firbase/messaging/firebasemessagingservice

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Startup the app
2. Run Push.SetEnabledAsync(true); in App.xaml.cs

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - e.g. Xamarin.Forms on all 3 platforms (Android, iOS and UWP).

      Android

2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
 .NET standard 2.0.
2. Which SDK version are you using?
    1.13.0
4. Which OS version did you experience the issue on?
   Android 8.1.0
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    Samsung Galaxy 8 Physical device
6. What third party libraries are you using? 
    CarouselView.FormsPlugin" Version="5.2.0"
Com.Airbnb.Xamarin.Forms.Lottie" Version="2.4.0" NoWarn="NU1701" 
Microsoft.AppCenter.Analytics" Version="1.12.0" 
Microsoft.AppCenter.Crashes" Version="1.12.0" 
microsoft.appcenter.push" Version="1.12.0" 
Microsoft.Graph" Version="1.12.0"
NLog" Version="4.5.11" 
Peerless.Forms" Version="1.0.5" 
Peerless.Logging" Version="1.0.5" 
Peerless.Security" Version="2.0.11" 
Peerless.Api.Client" Version="1.0.18" 
PeerlessWebDTOs" Version="1.2.14" 
Prism.Unity.Forms" Version="7.1.0.431" 
Refractored.XamForms.PullToRefresh" Version="2.2.3" 
SkiaSharp.Views.Forms" Version="1.68.0"
Syncfusion.Xamarin.SfCarousel" Version="16.3.0.29" 
Syncfusion.Xamarin.SfProgressBar" Version="16.3.0.29" 
Syncfusion.Xamarin.SfRadialMenu" Version="16.3.0.29" 
Syncfusion.Xamarin.SfTabView" Version="16.3.0.36" 
Syncfusion.Xamarin.SfTreeView" Version="16.3.0.36"
UXDivers.Effects" Version="0.6.3" 
Xam.Plugins.Forms.ImageCircle" Version="3.0.0.5" 
Xamarin.Essentials" Version="1.0.0" 
Xamarin.FFImageLoading" Version="2.4.4.859"
Xamarin.FFImageLoading.Forms" Version="2.4.4.859" 
Xamarin.FFImageLoading.Svg.Forms" Version="2.4.4.859" 
Xamarin.FFImageLoading.Transformations" Version="2.4.4.859" 
Xamarin.Forms" Version="3.4.0.1008975" 
UXDivers.Artina.Shared.Base" Version="2.7.1" 
UXDivers.Artina.Shared" Version="2.7.1"
Xamarin.Forms.CarouselView" Version="2.3.0-pre2"
Xamarin.Forms.GoogleMaps" Version="3.0.4" 
Xamarin.Forms.GoogleMaps.Bindings" Version="2.2.0" 
Xamarin.LiveReload" Version="0.3.27"
7. Please enable verbose logging for your app using `AppCenter.LogLevel = LogLevel.Verbose` before your call to `AppCenter.Start(...)` and include the logs here:

02-01 10:24:12.195 I/AppCenterCrashes(  746): Set up Xamarin crash handlers.Thread started: <Thread Pool> #2

02-01 10:24:12.205 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
02-01 10:24:12.205 D/Mono    (  746): Searching for 'java_interop_jnienv_new_object_array'.
02-01 10:24:12.205 D/Mono    (  746): Probing 'java_interop_jnienv_new_object_array'.
02-01 10:24:12.205 D/Mono    (  746): Found as 'java_interop_jnienv_new_object_array'.
02-01 10:24:12.205 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
Thread started: <Thread Pool> #302-01 10:24:12.205 D/Mono    (  746): Searching for 'java_interop_jnienv_set_object_array_element'.

02-01 10:24:12.205 D/Mono    (  746): Probing 'java_interop_jnienv_set_object_array_element'.
02-01 10:24:12.206 D/Mono    (  746): Found as 'java_interop_jnienv_set_object_array_element'.
02-01 10:24:12.209 I/AppCenter(  746): App Center SDK configured successfully.
02-01 10:24:12.210 D/AppCenter(  746): Cannot read instrumentation variables in a non-test environment.
02-01 10:24:12.211 I/chatty  (  746): uid=10578(u0_a578) com.peerless.xpeerience identical 1 line
02-01 10:24:12.212 D/AppCenter(  746): Cannot read instrumentation variables in a non-test environment.Thread started: <Thread Pool> #4

02-01 10:24:12.213 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
02-01 10:24:12.213 D/Mono    (  746): Searching for 'java_interop_jnienv_get_object_array_element'.
02-01 10:24:12.213 D/Mono    (  746): Probing 'java_interop_jnienv_get_object_array_element'.
02-01 10:24:12.213 D/Mono    (  746): Found as 'java_interop_jnienv_get_object_array_element'.
02-01 10:24:12.214 D/Mono    (  746): Assembly Ref addref Microsoft.AppCenter.Push[0x7096cb7180] -> Microsoft.AppCenter.Android.Bindings[0x7096cf1d00]: 6
02-01 10:24:12.218 D/AppCenter(  746): Loaded stored sessions: {1549034668953=1549034668953//1549034668950, 1549036139139=1549036139139//1549036139130, 1549036446723=1549036446723//1549036446717, 1549036992806=1549036992806//1549036992797, 1549037450607=1549037450607//1549037450602}
02-01 10:24:12.223 D/ConnectivityManager(  746): requestNetwork; CallingUid : 10578, CallingPid : 746Thread started: <Thread Pool> #5

02-01 10:24:12.234 I/AppCenter(  746): Changed maximum database size to 10485760 bytes.
02-01 10:24:12.234 D/AppCenter(  746): addGroup(group_core)
02-01 10:24:12.235 D/AppCenter(  746): Network available netId: 507
02-01 10:24:12.235 D/AppCenter(  746): Available networks netIds: [507]
02-01 10:24:12.236 D/AppCenter(  746): checkPendingLogs(group_core) pendingLogCount=0
02-01 10:24:12.237 D/AppCenter(  746): App Center initialized.
02-01 10:24:12.237 D/AppCenter(  746): removeGroup(group_analytics)
02-01 10:24:12.237 D/AppCenter(  746): removeGroup(group_analytics/one)
02-01 10:24:12.237 D/AppCenter(  746): addGroup(group_analytics)
02-01 10:24:12.238 D/AppCenter(  746): checkPendingLogs(group_analytics) pendingLogCount=0
02-01 10:24:12.238 D/AppCenter(  746): addGroup(group_analytics/one)
02-01 10:24:12.239 D/AppCenter(  746): checkPendingLogs(group_analytics/one) pendingLogCount=0
02-01 10:24:12.239 I/AppCenter(  746): Analytics service started from application.
02-01 10:24:12.240 D/AppCenter(  746): removeGroup(groupErrors)
02-01 10:24:12.240 D/AppCenter(  746): removeGroup(groupErrors/one)
02-01 10:24:12.240 D/AppCenter(  746): addGroup(groupErrors)
02-01 10:24:12.242 D/AppCenter(  746): checkPendingLogs(groupErrors) pendingLogCount=0
02-01 10:24:12.242 D/AppCenter(  746): addGroup(groupErrors/one)
02-01 10:24:12.243 D/AppCenter(  746): checkPendingLogs(groupErrors/one) pendingLogCount=0
02-01 10:24:12.249 D/AppCenterCrashes(  746): Processing crash report for the last session.
02-01 10:24:12.269 D/Mono    (  746): Assembly Ref addref Microsoft.AppCenter.Crashes[0x7096cf1c80] -> Microsoft.AppCenter.Android.Bindings[0x7096cf1d00]: 7
02-01 10:24:12.300 D/AppCenterCrashes(  746): Processed crash report for the last session.
02-01 10:24:12.300 D/AppCenterCrashes(  746): Process pending error file: /data/user/0/com.peerless.xpeerience/files/error/597011f8-d52e-40bf-93a0-51d8110f3d7d.json
Thread started:  #6
02-01 10:24:12.361 D/AppCenterCrashes(  746): Process pending error file: /data/user/0/com.peerless.xpeerience/files/error/17593712-d557-4b9a-818a-ecd9dbafa7a5.json
02-01 10:24:12.399 D/AppCenterCrashes(  746): Process pending error file: /data/user/0/com.peerless.xpeerience/files/error/5ee6628b-1c15-49d5-aa33-c5f50646984e.json
02-01 10:24:12.436 D/AppCenterCrashes(  746): Process pending error file: /data/user/0/com.peerless.xpeerience/files/error/d7c805fb-87d7-46e2-91b2-c12d49c90b9c.json
02-01 10:24:12.462 I/zygote64(  746): Do partial code cache collection, code=30KB, data=22KB
02-01 10:24:12.462 I/zygote64(  746): After code cache collection, code=30KB, data=22KB
02-01 10:24:12.462 I/zygote64(  746): Increasing code cache capacity to 128KB
02-01 10:24:12.472 D/AppCenterCrashes(  746): Process pending error file: /data/user/0/com.peerless.xpeerience/files/error/21014049-e492-450e-a042-b31a26335e65.jsonThread started: <Thread Pool> #7

02-01 10:24:12.480 I/AppCenter(  746): Crashes service started from application.
02-01 10:24:12.480 D/AppCenter(  746): removeGroup(group_push)
02-01 10:24:12.480 D/AppCenter(  746): removeGroup(group_push/one)
02-01 10:24:12.480 D/AppCenter(  746): addGroup(group_push)
02-01 10:24:12.481 D/Mono    (  746): Assembly Ref addref Microsoft.AppCenter.Analytics[0x7096cf1a00] -> Microsoft.AppCenter.Android.Bindings[0x7096cf1d00]: 8
02-01 10:24:12.481 D/Mono    (  746): Assembly Ref addref Microsoft.AppCenter.Analytics[0x7096cf1a00] -> Mono.Android[0x7096d96b00]: 66
02-01 10:24:12.482 D/AppCenter(  746): checkPendingLogs(group_push) pendingLogCount=0
02-01 10:24:12.482 D/AppCenter(  746): addGroup(group_push/one)
02-01 10:24:12.483 D/AppCenter(  746): checkPendingLogs(group_push/one) pendingLogCount=0
02-01 10:24:12.483 D/Mono    (  746): Assembly Ref addref Xamarin.Live.Reload[0x7096d96a00] -> Xamarin.Forms.Core[0x7096d7ca00]: 36
02-01 10:24:12.487 E/AndroidRuntime(  746): FATAL EXCEPTION: AppCenter.Looper
02-01 10:24:12.487 E/AndroidRuntime(  746): Process: com.peerless.xpeerience, PID: 746
Thread started: <Thread Pool> #802-01 10:24:12.487 E/AndroidRuntime(  746): java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/messaging/FirebaseMessagingService;

02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.push.FirebaseUtils.setFirebaseMessagingServiceEnabled(FirebaseUtils.java:65)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.push.Push.registerPushToken(Push.java:429)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.push.Push.applyEnabledState(Push.java:259)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.AbstractAppCenterService.onStarted(AbstractAppCenterService.java:201)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.push.Push.onStarted(Push.java:291)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.AppCenter.finishStartServices(AppCenter.java:897)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.AppCenter.access$600(AppCenter.java:51)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at com.microsoft.appcenter.AppCenter$8.run(AppCenter.java:819)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at android.os.Handler.handleCallback(Handler.java:789)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at android.os.Handler.dispatchMessage(Handler.java:98)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at android.os.Looper.loop(Looper.java:164)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at android.os.HandlerThread.run(HandlerThread.java:65)
02-01 10:24:12.487 E/AndroidRuntime(  746): Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.messaging.FirebaseMessagingService" on path: DexPathList[[zip file "/data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/lib/arm64, /data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
02-01 10:24:12.487 E/AndroidRuntime(  746): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
Loaded assembly: System.Diagnostics.TraceSource.dll [External]02-01 10:24:12.487 E/AndroidRuntime(  746): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)

02-01 10:24:12.487 E/AndroidRuntime(  746): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
02-01 10:24:12.487 E/AndroidRuntime(  746): 	... 12 moreLoaded assembly: System.Collections.Concurrent.dll [External]

02-01 10:24:12.488 D/Mono    (  746): Assembly Ref addref Xamarin.Live.Reload[0x7096d96a00] -> System.Net.Mqtt[0x7096d45580]: 2
02-01 10:24:12.499 D/AppCenterCrashes(  746): Saving uncaught exception.
02-01 10:24:12.507 D/Mono    (  746): Assembly Ref addref Xamarin.Live.Reload[0x7096d96a00] -> Xamarin.Live[0x7096d96300]: 2
02-01 10:24:12.511 D/Mono    (  746): DllImport attempting to load: '/system/lib64/libc.so'.Loaded assembly: System.Linq.dll [External]

02-01 10:24:12.511 D/Mono    (  746): DllImport loaded library '/system/lib64/libc.so'.
02-01 10:24:12.511 D/Mono    (  746): DllImport searching in: '/system/lib64/libc.so' ('/system/lib64/libc.so').
02-01 10:24:12.511 D/Mono    (  746): Searching for 'uname'.
02-01 10:24:12.511 D/Mono    (  746): Probing 'uname'.
02-01 10:24:12.511 D/Mono    (  746): Found as 'uname'.
02-01 10:24:12.516 D/Mono    (  746): Assembly Ref addref Mono.Android[0x7096d96b00] -> System[0x7081507680]: 17
02-01 10:24:12.516 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
02-01 10:24:12.517 D/Mono    (  746): Searching for '_monodroid_getifaddrs'.
02-01 10:24:12.517 D/Mono    (  746): Probing '_monodroid_getifaddrs'.
02-01 10:24:12.517 D/Mono    (  746): Found as '_monodroid_getifaddrs'.
02-01 10:24:12.518 D/Mono    (  746): DllImport attempting to load: '__Internal'.
02-01 10:24:12.518 D/Mono    (  746): DllImport loaded library '(null)'.
02-01 10:24:12.518 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
02-01 10:24:12.518 D/Mono    (  746): Searching for '_monodroid_get_android_api_level'.
02-01 10:24:12.518 D/Mono    (  746): Probing '_monodroid_get_android_api_level'.
02-01 10:24:12.518 D/Mono    (  746): Found as '_monodroid_get_android_api_level'.Loaded assembly: System.Runtime.Extensions.dll [External]

02-01 10:24:12.523 D/Mono    (  746): DllImport searching in: '__Internal' ('(null)').
02-01 10:24:12.523 D/Mono    (  746): Searching for '_monodroid_freeifaddrs'.
02-01 10:24:12.523 D/Mono    (  746): Probing '_monodroid_freeifaddrs'.
02-01 10:24:12.523 D/Mono    (  746): Found as '_monodroid_freeifaddrs'.
02-01 10:24:12.529 D/AppCenterCrashes(  746): Saved JSON content for ingestion into /data/user/0/com.peerless.xpeerience/files/error/69a91b17-0d1c-427f-ad8c-37f65d807594.json
02-01 10:24:12.541 D/AppCenterCrashes(  746): Saved Throwable as is for client side inspection in /data/user/0/com.peerless.xpeerience/files/error/69a91b17-0d1c-427f-ad8c-37f65d807594.throwable throwable:
02-01 10:24:12.541 D/AppCenterCrashes(  746): java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/messaging/FirebaseMessagingService;
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.push.FirebaseUtils.setFirebaseMessagingServiceEnabled(FirebaseUtils.java:65)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.push.Push.registerPushToken(Push.java:429)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.push.Push.applyEnabledState(Push.java:259)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.AbstractAppCenterService.onStarted(AbstractAppCenterService.java:201)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.push.Push.onStarted(Push.java:291)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.AppCenter.finishStartServices(AppCenter.java:897)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.AppCenter.access$600(AppCenter.java:51)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at com.microsoft.appcenter.AppCenter$8.run(AppCenter.java:819)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at android.os.Handler.handleCallback(Handler.java:789)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at android.os.Handler.dispatchMessage(Handler.java:98)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at android.os.Looper.loop(Looper.java:164)Thread started:  #9
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at android.os.HandlerThread.run(HandlerThread.java:65)

02-01 10:24:12.541 D/AppCenterCrashes(  746): Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.messaging.FirebaseMessagingService" on path: DexPathList[[zip file "/data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/lib/arm64, /data/app/com.peerless.xpeerience-qcB58tqz3RKqBMkcO8ydPQ==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
02-01 10:24:12.541 D/AppCenterCrashes(  746): 	... 12 more
02-01 10:24:12.605 D/Mono    (  746): Assembly Ref addref Xamarin.Live.Reload[0x7096d96a00] -> Merq[0x7096cf1900]: 2Loaded assembly: System.Threading.dll [External]

**Comment by guperrot** [comment]
Hi,

Are you using Proguard? If yes did you configure it like in our docs?
Can you share the AndroidManifes.xml, packages.config (if any, or project.json if any) and the csproj files of the Android specific project? (not the portable one).

**Comment by chrisevans9629** [comment]
## Pro guard:
I am not using pro guard

## AndroidManifest:
`<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:installLocation="preferExternal" package="com.peerless.xpeerience" android:versionName="2.0" android:versionCode="10">
	<uses-sdk android:minSdkVersion="17" android:targetSdkVersion="27" />
	<application android:icon="@drawable/icon" android:label="Peerless" android:largeHeap="@bool/largeheap" >
	  <receiver android:name="com.google.firebase.iid.FirebaseInstanceIdInternalReceiver" android:exported="false" />
	  <receiver android:name="com.google.firebase.iid.FirebaseInstanceIdReceiver" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND">
	    <intent-filter>
	      <action android:name="com.google.android.c2dm.intent.RECEIVE" />
	      <action android:name="com.google.android.c2dm.intent.REGISTRATION" />
	      <category android:name="${applicationId}" />
	    </intent-filter>
	  </receiver>
	</application>
	<uses-permission android:name="android.permission.INTERNET" />
  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
</manifest>`
## app.config:
`<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-12.0.0.0" newVersion="12.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="WebSocket4Net" publicKeyToken="eb4e154b696bf72a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-0.15.1.10" newVersion="0.15.1.10" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="SuperSocket.ClientEngine" publicKeyToken="ee9af13f57f00acc" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-0.9.0.0" newVersion="0.9.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Prism" publicKeyToken="40ee6c3a2184dc59" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-7.1.0.431" newVersion="7.1.0.431" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reactive.Interfaces" publicKeyToken="94bc3704cddfc263" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.0.0.0" newVersion="4.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Reactive.Linq" publicKeyToken="94bc3704cddfc263" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.0.6000.0" newVersion="3.0.6000.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>`
## packages:
We use the new project packing system.  so the packages are below:
### nuget:
`<ItemGroup>
    <PackageReference Include="Com.Airbnb.Xamarin.Forms.Lottie">
      <Version>2.4.0</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.AppCenter.Analytics">
      <Version>1.13.0</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.AppCenter.Crashes">
      <Version>1.13.0</Version>
    </PackageReference>
    <PackageReference Include="Newtonsoft.Json">
      <Version>11.0.2</Version>
    </PackageReference>
    <PackageReference Include="Prism.Unity.Forms">
      <Version>7.1.0.431</Version>
    </PackageReference>
    <PackageReference Include="Refractored.XamForms.PullToRefresh">
      <Version>2.2.3</Version>
    </PackageReference>
    <PackageReference Include="SkiaSharp.Views.Forms">
      <Version>1.68.0</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfCarousel">
      <Version>16.3.0.29</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfProgressBar">
      <Version>16.3.0.29</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfTabView">
      <Version>16.3.0.36</Version>
    </PackageReference>
    <PackageReference Include="System.Reactive.Interfaces">
      <Version>4.0.0-preview.2.build.379</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.Artina.Shared">
      <Version>2.7.1</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.Effects">
      <Version>0.6.3</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.GorillaPlayer.SDK">
      <Version>1.2.0</Version>
    </PackageReference>
    <PackageReference Include="Xam.Plugins.Forms.ImageCircle">
      <Version>3.0.0.5</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Core.Common">
      <Version>1.0.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Lifecycle.Common">
      <Version>1.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Lifecycle.Runtime">
      <Version>1.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Animated.Vector.Drawable">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Annotations">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Compat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Core.UI">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Core.Utils">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Design">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Exif">
      <Version>25.1.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Fragment">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Media.Compat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Transition">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v17.Leanback">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v4">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.AppCompat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.CardView">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.MediaRouter">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.Palette">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.RecyclerView">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Vector.Drawable">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Build.Download">
      <Version>0.4.9</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Essentials">
      <Version>1.0.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Forms">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Svg.Forms">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Transformations">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Forms">
      <Version>3.4.0.1008975</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Forms.GoogleMaps">
      <Version>3.0.4</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Base">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Basement">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Maps">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Tasks">
      <Version>60.1142.1</Version>
    </PackageReference>
  </ItemGroup>`

### DLLs:
`<ItemGroup>
    <Reference Include="Java.Interop" />
    <Reference Include="System" />
    <Reference Include="System.IO.Compression" />
    <Reference Include="System.Net.Http" />
    <Reference Include="System.Xml" />
    <Reference Include="System.Core" />
    <Reference Include="Mono.Android" />
    <Reference Include="mscorlib" />
    <Reference Include="System.Xml.Linq" />
  </ItemGroup>`

## csproj file:
`<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="..\packages\Xamarin.Forms.2.5.0.280555\build\netstandard1.0\Xamarin.Forms.props" Condition="Exists('..\packages\Xamarin.Forms.2.5.0.280555\build\netstandard1.0\Xamarin.Forms.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectTypeGuids>{EFBA0AD7-5A72-4C68-AF49-83D382785DCF};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
    <ProjectGuid>{B6C3195C-0075-436E-826B-FEF49284DD32}</ProjectGuid>
    <OutputType>Library</OutputType>
    <RootNamespace>Peerless.Expeerience.Droid</RootNamespace>
    <AndroidApplication>True</AndroidApplication>
    <AndroidResgenFile>Resources\Resource.designer.cs</AndroidResgenFile>
    <AndroidResgenClass>Resource</AndroidResgenClass>
    <MonoAndroidResourcePrefix>Resources</MonoAndroidResourcePrefix>
    <MonoAndroidAssetsPrefix>Assets</MonoAndroidAssetsPrefix>
    <AndroidManifest>Properties\AndroidManifest.xml</AndroidManifest>
    <NuGetPackageImportStamp>
    </NuGetPackageImportStamp>
    <TargetFrameworkVersion>v8.1</TargetFrameworkVersion>
    <AssemblyName>Peerless.Expeerience.Droid</AssemblyName>
    <ReleaseVersion>2.0</ReleaseVersion>
    <AndroidTlsProvider>
    </AndroidTlsProvider>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug</OutputPath>
    <DefineConstants>DEBUG;</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <AndroidLinkMode>SdkOnly</AndroidLinkMode>
    <EmbedAssembliesIntoApk>false</EmbedAssembliesIntoApk>
    <AndroidSupportedAbis>x86;armeabi;armeabi-v7a;x86_64;arm64-v8a</AndroidSupportedAbis>
    <IntermediateOutputPath>C:\VS Short Path</IntermediateOutputPath>
    <AotAssemblies>false</AotAssemblies>
    <EnableLLVM>false</EnableLLVM>
    <BundleAssemblies>false</BundleAssemblies>
    <AndroidEnableMultiDex>true</AndroidEnableMultiDex>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <DebugType>
    </DebugType>
    <Optimize>true</Optimize>
    <OutputPath>bin\Release</OutputPath>
    <DefineConstants>PUBLISH_STORE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <AndroidUseSharedRuntime>false</AndroidUseSharedRuntime>
    <AndroidSupportedAbis>armeabi;armeabi-v7a;x86;arm64-v8a;x86_64</AndroidSupportedAbis>
    <IntermediateOutputPath>C:\VS Short Path</IntermediateOutputPath>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Gorilla|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Gorilla</OutputPath>
    <DefineConstants>DEBUG; GORILLA</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
    <AndroidLinkMode>None</AndroidLinkMode>
    <EmbedAssembliesIntoApk>false</EmbedAssembliesIntoApk>
    <AotAssemblies>false</AotAssemblies>
    <EnableLLVM>false</EnableLLVM>
    <BundleAssemblies>false</BundleAssemblies>
    <AndroidEnableMultiDex>true</AndroidEnableMultiDex>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="Java.Interop" />
    <Reference Include="System" />
    <Reference Include="System.IO.Compression" />
    <Reference Include="System.Net.Http" />
    <Reference Include="System.Xml" />
    <Reference Include="System.Core" />
    <Reference Include="Mono.Android" />
    <Reference Include="mscorlib" />
    <Reference Include="System.Xml.Linq" />
  </ItemGroup>
  <ItemGroup>
    <Compile Include="AppWidget.cs" />
    <Compile Include="Helpers\AndroidHelpers.cs" />
    <Compile Include="MainActivity.cs" />
    <Compile Include="MyApp.cs" />
    <Compile Include="Renderers\BotomNavTabPageRenderer.cs" />
    <Compile Include="Resources\Resource.designer.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
  </ItemGroup>
  <ItemGroup>
    <None Include="app.config" />
    <EmbeddedResource Include="GrialLicense" />
    <GoogleServicesJson Include="google-services.json" />
    <None Include="Resources\AboutResources.txt" />
    <None Include="Properties\AndroidManifest.xml">
      <SubType>Designer</SubType>
    </None>
    <None Include="Assets\AboutAssets.txt" />
    <None Include="Assets\fontawesome-webfont.ttf" />
    <None Include="Assets\ionicons.ttf" />
    <AndroidResource Include="Resources\drawable\building.json" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\icon.png" />
    <AndroidResource Include="Resources\drawable\splash.png" />
    <AndroidResource Include="Resources\drawable\hamburguer_icon.png" />
    <AndroidResource Include="Resources\values\Colors.xml" />
    <AndroidResource Include="Resources\drawable\walkthrough_generic_phone_bg.png" />
    <AndroidResource Include="Resources\drawable\welcome_bg.jpg" />
    <AndroidResource Include="Resources\drawable\social_header_bg_0.jpg" />
    <AndroidResource Include="Resources\drawable\social_header_bg_1.jpg" />
    <AndroidResource Include="Resources\drawable\user_profile_1.jpg" />
    <AndroidResource Include="Resources\drawable\user_profile_0.jpg" />
    <AndroidResource Include="Resources\drawable\signup_bg.jpg" />
    <AndroidResource Include="Resources\drawable\logo.png" />
    <AndroidResource Include="Resources\values-v19\bools.xml" />
    <AndroidResource Include="Resources\values\Style.xml" />
    <AndroidResource Include="Resources\layout\Toolbar.axml" />
    <AndroidResource Include="Resources\layout\Tabs.axml" />
    <AndroidResource Include="Resources\values-v21\Style.xml" />
    <AndroidResource Include="Resources\drawable\welcome_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable\signup_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable\splash_drawable.xml" />
    <AndroidResource Include="Resources\drawable\pass_recovery_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable\pass_recovery_bg.jpg" />
    <AndroidResource Include="Resources\drawable\generic_bg_image.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\generic_bg_image.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\hamburguer_icon.png" />
    <AndroidResource Include="Resources\drawable-hdpi\icon.png" />
    <AndroidResource Include="Resources\drawable-hdpi\login_bg.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\login_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\logo.png" />
    <AndroidResource Include="Resources\drawable-hdpi\pass_recovery_bg.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\pass_recovery_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\signup_bg.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\signup_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\social_header_bg_0.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\social_header_bg_1.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\splash.png" />
    <AndroidResource Include="Resources\drawable-hdpi\user_profile_0.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\user_profile_1.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\walkthrough_generic_phone_bg.png" />
    <AndroidResource Include="Resources\drawable-hdpi\welcome_bg.jpg" />
    <AndroidResource Include="Resources\drawable-hdpi\welcome_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\generic_bg_image.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\hamburguer_icon.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\icon.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\login_bg.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\login_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\logo.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\pass_recovery_bg.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\pass_recovery_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\signup_bg.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\signup_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\social_header_bg_0.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\social_header_bg_1.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\splash.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\user_profile_0.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\user_profile_1.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\walkthrough_generic_phone_bg.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\welcome_bg.jpg" />
    <AndroidResource Include="Resources\drawable-xhdpi\welcome_bg_tablet.jpg" />
    <AndroidResource Include="Resources\drawable-xxhdpi\hamburguer_icon.png" />
    <AndroidResource Include="Resources\drawable-xxhdpi\icon.png" />
    <AndroidResource Include="Resources\drawable-xxhdpi\logo.png" />
    <AndroidResource Include="Resources\drawable-xxhdpi\splash.png" />
  </ItemGroup>
  <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
  <ItemGroup>
    <AndroidAsset Include="Assets\grialshapes.ttf" />
    <AndroidAsset Include="Assets\grial_animation.json" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\Peerless.Expeerience\Peerless.Expeerience.csproj">
      <Project>{7041522B-6406-4AA7-8C6D-8AD1661DD392}</Project>
      <Name>Peerless.Expeerience</Name>
    </ProjectReference>
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\login_bg.jpg" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\login_bg_tablet.jpg" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\calendar_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\profile_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\calendar_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\home_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\profile_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\tracker_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\tv_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\tv_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\document_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\communication_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\completed_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\completed_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\communication_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\document_event_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\messages_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Com.Airbnb.Xamarin.Forms.Lottie">
      <Version>2.4.0</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.AppCenter.Analytics">
      <Version>1.13.0</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.AppCenter.Crashes">
      <Version>1.13.0</Version>
    </PackageReference>
    <PackageReference Include="Newtonsoft.Json">
      <Version>11.0.2</Version>
    </PackageReference>
    <PackageReference Include="Prism.Unity.Forms">
      <Version>7.1.0.431</Version>
    </PackageReference>
    <PackageReference Include="Refractored.XamForms.PullToRefresh">
      <Version>2.2.3</Version>
    </PackageReference>
    <PackageReference Include="SkiaSharp.Views.Forms">
      <Version>1.68.0</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfCarousel">
      <Version>16.3.0.29</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfProgressBar">
      <Version>16.3.0.29</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfTabView">
      <Version>16.3.0.36</Version>
    </PackageReference>
    <PackageReference Include="System.Reactive.Interfaces">
      <Version>4.0.0-preview.2.build.379</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.Artina.Shared">
      <Version>2.7.1</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.Effects">
      <Version>0.6.3</Version>
    </PackageReference>
    <PackageReference Include="UXDivers.GorillaPlayer.SDK">
      <Version>1.2.0</Version>
    </PackageReference>
    <PackageReference Include="Xam.Plugins.Forms.ImageCircle">
      <Version>3.0.0.5</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Core.Common">
      <Version>1.0.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Lifecycle.Common">
      <Version>1.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Arch.Lifecycle.Runtime">
      <Version>1.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Animated.Vector.Drawable">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Annotations">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Compat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Core.UI">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Core.Utils">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Design">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Exif">
      <Version>25.1.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Fragment">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Media.Compat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Transition">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v17.Leanback">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v4">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.AppCompat">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.CardView">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.MediaRouter">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.Palette">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.v7.RecyclerView">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Vector.Drawable">
      <Version>27.0.2</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Build.Download">
      <Version>0.4.9</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Essentials">
      <Version>1.0.0</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Forms">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Svg.Forms">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Transformations">
      <Version>2.4.4.859</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Forms">
      <Version>3.4.0.1008975</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Forms.GoogleMaps">
      <Version>3.0.4</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Base">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Basement">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Maps">
      <Version>60.1142.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Tasks">
      <Version>60.1142.1</Version>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-ldpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-ldpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\welcome.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\Residence_Inn5.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\events.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\home.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\messages.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\profile.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\projects.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\team.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\tracker.PNG" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\ic_launcher.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\microsoft_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\microsoft_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <EmbeddedResource Include="Resources\drawable\peerless_logo.svg" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\quote_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\quote_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\quote_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\quote_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\quote_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\order_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\order_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\order_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\order_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\order_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\factory.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\delivery_truck_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\forklift_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\peerlesslogo.svg" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\delivery_truck_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\forklift_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\delivery_truck_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\forklift_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\forklift_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\forklift_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\home_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\help_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\search_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\help_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\search_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\help_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\search_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\help_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\search_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\help_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\search_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\share_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\share_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\share_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\share_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\share_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\home_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\home_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\home_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\schedule_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\schedule_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\schedule_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\schedule_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\schedule_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\messages_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\tracker_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\messages_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\tracker_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\tracker_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\messages_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\messages_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\tracker_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\production_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\production_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\production_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\delivery_truck_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\production_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\delivery_truck_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\production_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\information_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\television_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\information_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\television_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\information_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\television_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\information_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\television_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\information_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\library_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\scorecard_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\service_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\settings_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\television_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\ttf_xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\ttf_xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\ttf_xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\ttf_xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\ttf_xpeerience_logo.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\layout\Widget.axml">
      <SubType>Designer</SubType>
    </AndroidResource>
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\xml\appwidgetprovider.xml" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\announcement.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\android.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\previewImage.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\values\Strings.xml" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable\delivery_truck_icon.png">
      <Generator>MSBuild:UpdateGeneratedFiles</Generator>
      <SubType>Designer</SubType>
    </AndroidResource>
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\heart_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-mdpi\heart_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xhdpi\heart_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxhdpi\heart_icon.png" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-xxxhdpi\heart_icon.png" />
  </ItemGroup>
  <Import Project="..\packages\Xamarin.Forms.2.5.0.280555\build\netstandard1.0\Xamarin.Forms.targets" Condition="Exists('..\packages\Xamarin.Forms.2.5.0.280555\build\netstandard1.0\Xamarin.Forms.targets')" />
</Project>`

**Comment by guperrot** [comment]
The Push nuget is missing in the Android csproj (you probably have it only in the portable project), please install Microsoft.AppCenter.Push in the Android specific project. That's why symbols are missing as the portable implementation (which is empty) is used instead of the Android one.

After that you probably don't have the GoogleServicesJson as a build action on the google-services.json file as this action is available only after installing the missing package, please refer to https://docs.microsoft.com/en-us/appcenter/sdk/push/xamarin-android#3-add-google-servicesjson to double check that step.

**Comment by chrisevans9629** [comment]
@guperrot Ah that's it, thank you!  I had the GoogleServicesJson.  It looks like the reference was removed during a git merge.

**Comment by guperrot** [comment]
No problem and thanks for using AppCenter.

**Comment by dfoulk** [comment]
I've encountered this exception a few times recently...

> Unhandled Exception: Java.Lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/messaging/FirebaseMessagingService; occurred

To remedy this, I have to:

1. Delete all **obj** and **bin** folders in the solution
2. Clear NuGet cache
3. Restart Visual Studio
4. Build my app again

And... voila- no more exception. @guperrot I have everything configured properly, however I am a couple of releases behind (I'm using **Microsoft.AppCenter.*** **v1.12.0**).

Any idea why my little to-do list fixes the exception (temporary as it may/may not be)?

**Comment by jwallra** [comment]
Hi @dfoulk - Since this is a new issue I've created a new issue using the "reference in new issue" feature (that I just now discovered!).  I'll investigate and respond there! #851

**Comment by dfoulk** [comment]
@jwallra - I didn't know you could do that- sweet! Thanks for checking this out!
